### PR TITLE
Fix base_url and Fix for video not able to play or download

### PR DESCRIPTION
**Pull request by RaynardGerraldo**
Trace gogoanime's base url "https://gogoanime.cm" redirects and getting the final url from redirect as base_url value

### CODE REVIEW

File: ani-cli [hunk 1]
```diff
@@ -6,7 +6,7 @@ player_fn="mpv"
 
 prog="ani-cli"
 logfile="${XDG_CACHE_HOME:-$HOME/.cache}/ani-hsts"
-base_url="https://www1.gogoanime.cm"
+base_url=$(curl -s -L -o /dev/null -w "%{url_effective}\n" https://gogoanime.cm)
```
Comment (Kungger-git) [inline ani-cli:9]: This honestly resolves the problem for everyone that gets no results when trying to search for something

**Comment by shulkk** [comment]
Thanks.

**Review by shulkk — APPROVED** [review]

**Comment by RaynardGerraldo** [comment]
> Thanks.

np, thanks for the approval🙂

File: ani-cli (hunk 1 above)
Comment (Nikson230) [reply ani-cli:9]: Thanks

**Review by solarft — APPROVED** [review]

**Comment by ScaroDj** [comment]
Hello, no longer a fix, I had "ww10.gogoanimehub.tv" and now it is "gogoanimeapp.com", they have always and they will continue to bounce the domain names around. It's the way they evade copyright holders. 

Could there be a way to resolve via DNS to the new URL or even tell the user to resolve it using a web browser and store them in a config file?

**Comment by RaynardGerraldo** [comment]
> Hello, no longer a fix, I had "ww10.gogoanimehub.tv" and now it is "gogoanimeapp.com", they have always and they will continue to bounce the domain names around. It's the way they evade copyright holders.
> 
> Could there be a way to resolve via DNS to the new URL or even tell the user to resolve it using a web browser and store them in a config file?

Did you use (https://gogoanime.cm) for the starting url like in the fix? I just reran the script and it still works and leads me to www2.gogoanime.cm,both from the browser and the script

**Comment by RaynardGerraldo** [comment]
5a0261126682295387977509066f8a6f0ead30f3 Temporary fix for video not able to play or download

**Comment by OnionHat** [comment]
If this PR goes through `lynx` should be added as one of the dependencies in README.

**Review by ura43 — APPROVED** [review]
The proposed changes look good to me

**Review by ura43 — CHANGES_REQUESTED** [review]
Please get rid of the lynx requirement and also catch errors for vlc

**Comment by RaynardGerraldo** [comment]
> Please get rid of the lynx requirement and also catch errors for vlc

on it

**Comment by ura43** [comment]
> > Please get rid of the lynx requirement and also catch errors for vlc
> 
> on it

Thanks for tackling this issue. In its current form I still have the same issue. Tried to watch flcl progressive ep 1 on debian unstable. In #231 another user appears to reproduce what I see. 
It definitely says "Video Played", even though it shouldn't

**Comment by RaynardGerraldo** [comment]
> flcl progres

thats good to hear, i removed lynx dependency, vlc error message might take a while to figure out, since it opens vlc even if there is no video to play, havent found a way to output the message before it runs

**Comment by ura43** [comment]
Thats good news. As for vlc, don't bother bending over backwards to make the error display without vlc launching. I'm sure users will close vlc on their own if nothing plays and as long as the error comes after that its good enough.
Wget is already a huge improvement over lynx. If it is not too troublesome, could you adapt it to curl so we have zero new dependencies?
wget and curl are two sides of the same coin iirc so shouldn't be too difficult

**Review by ura43 — CHANGES_REQUESTED** [review]
Great change, but pls investigate if curl instead of wget is practical to avoid new dependency

**Comment by ura43** [comment]
As explained in [this](https://unix.stackexchange.com/questions/593023/equivalent-curl-command-from-wget-command) stackoverflow post, you can just replace `wget -q0-` with `curl -s`

**Comment by ura43** [comment]
Merging and adding the wget to curl substitution right after

**Comment by RaynardGerraldo** [comment]
> Please get rid of the lynx requirement and also catch errors for vlc

ok ill try

> As explained in [this](https://unix.stackexchange.com/questions/593023/equivalent-curl-command-from-wget-command) stackoverflow post, you can just replace `wget -q0-` with `curl -s`



> Thats good news. As for vlc, don't bother bending over backwards to make the error display without vlc launching. I'm sure users will close vlc on their own if nothing plays and as long as the error comes after that its good enough. Copy pasting the stuff you added in the mpv case to the vlc case should do the trick. Wget is already a huge improvement over lynx. If it is not too troublesome, could you adapt it to curl so we have zero new dependencies? wget and curl are two sides of the same coin iirc so shouldn't be too difficult.



> Thats good news. As for vlc, don't bother bending over backwards to make the error display without vlc launching. I'm sure users will close vlc on their own if nothing plays and as long as the error comes after that its good enough. Copy pasting the stuff you added in the mpv case to the vlc case should do the trick. Wget is already a huge improvement over lynx. If it is not too troublesome, could you adapt it to curl so we have zero new dependencies? wget and curl are two sides of the same coin iirc so shouldn't be too difficult.

ok , ill try and do that for vlc. I didnt realize that wget is still an extra dependency, should i open a new pr after this?

**Pull request merged**